include/opw_kinematics/opw_parameters.h

```diff
@@ -9,16 +9,16 @@ namespace opw_kinematics
 template <typename T>
 struct Parameters
 {
-  static_assert(std::is_floating_point<T>::value,
-                "OPW parameters must be templatized with floating point type");
+//  static_assert(std::is_floating_point<T>::value,
+//                "OPW parameters must be templatized with floating point type");
```

Comment:
Here fi you could provide a specialisation for half-float returning `true` for `std::is_floating_point<your_half_float_type>` instead of disabling this assert. It's there for a reason and it'd be unfortunate if we had to remove it completely.

Reply:
Ok, so that would've been nice, if it were allowed. [This seems to suggest it is not](https://stackoverflow.com/questions/30020005).

Reply:
So basically if a user provided a template specialization of `std::is_floating_point` in their user code that creates an instance of `opw_kinematics`, it would compile and function properly but would violate the C++ standard?

Reply:
> So basically if a user provided a template specialization of `std::is_floating_point` in their user code that creates an instance of `opw_kinematics`, it would compile and function properly but would violate the C++ standard?

yes. I realise that might seem strange, but being in compliance and your code compiling are two different things (GCC used to have quite a few extensions/quirks that made it somewhat not in compliance or go out-of-spec in the earlier days).

A quick look at the `half` library (header really) suggests there is already support for quite a few C++11 traits, so either this is an oversight or the author purposefully left this one out.

Might be worth a quick email / post on his tracker.